src/core/src/queue.rs

```diff
@@ -58,19 +58,51 @@ impl<Q: CommandQueue> DerefMut for GeneralQueue<Q> {
     }
 }
 
-impl<Q: CommandQueue> Into<GraphicsQueue<Q>> for GeneralQueue<Q> {
-    fn into(self) -> GraphicsQueue<Q> {
-        GraphicsQueue(self.0)
+impl<Q: CommandQueue> From<GeneralQueue<Q>> for GraphicsQueue<Q> {
+    fn from(queue: GeneralQueue<Q>) -> GraphicsQueue<Q> {
+        GraphicsQueue(queue.0)
     }
 }
-impl<Q: CommandQueue> Into<ComputeQueue<Q>> for GeneralQueue<Q> {
-    fn into(self) -> ComputeQueue<Q> {
-        ComputeQueue(self.0)
+impl<Q: CommandQueue> From<GeneralQueue<Q>> for ComputeQueue<Q> {
+    fn from(queue: GeneralQueue<Q>) -> ComputeQueue<Q> {
+        ComputeQueue(queue.0)
     }
 }
-impl<Q: CommandQueue> Into<TransferQueue<Q>> for GeneralQueue<Q> {
-    fn into(self) -> TransferQueue<Q> {
-        TransferQueue(self.0)
+impl<Q: CommandQueue> From<GeneralQueue<Q>> for TransferQueue<Q> {
+    fn from(queue: GeneralQueue<Q>) -> TransferQueue<Q> {
+        TransferQueue(queue.0)
+    }
+}
+
+impl<'a, Q: CommandQueue> From<&'a GeneralQueue<Q>> for &'a GraphicsQueue<Q> {
+    fn from(queue: &'a GeneralQueue<Q>) -> &'a GraphicsQueue<Q> {
+        unsafe { &*(queue as *const _ as *const GraphicsQueue<Q>) }
```

Comment:
Right.. this only made sense for the vulkan backend..
I'm refactoring this towards something like `SpecificQueueBase<Q: CommandQueue, R: Repr<Q>>(R, PhantomData<Q>>` which should allow to deal with it in a safer way!